### PR TITLE
Adding Windows Salt Capabilities

### PR DESCRIPTION
**Pull request by KevinPlus3**
When Watchmaker is run on Windows, it will now install Salt silently and set up minion configuration files and run salt tasks.  Refectored base.py since both Managers use the same functionality.

---
This change is [Reviewable](https://reviewable.io/reviews/plus3it/watchmaker/43)

### CODE REVIEW

File: src/watchmaker/managers/base.py [hunk 1]
```diff
@@ -242,23 +159,13 @@ def download_file(self, url, filename, sourceiss3bucket=False):
...
-
-        if rsp != 0:
-            exceptionhandler('Command failed.\n{0}'.format(str(cmd)))
-
+    @abc.abstractmethod
     def create_working_dir(self, basedir, prefix):
         """
-        Create a directory in `basedir` with a prefix of `prefix`.
 
-        Args:
```
Comment (lorengordon) [inline src/watchmaker/managers/base.py]: Ahh, while you were out, we started changing to google-style docstrings, like this one....

Comment (KevinPlus3) [inline src/watchmaker/managers/base.py]: Ah, okay.  I will convert all of them to google-style docstrings tomorrow morning then including whatever Linux ones have not been changed yet.

Comment (lorengordon) [inline src/watchmaker/managers/base.py]: Let's do that in a new PR. We can merge this. I'd also like to switch the print statements to log functions...

**Comment by lorengordon** [comment]
@MarionTheBull, can you review this as well?

File: src/watchmaker/managers/base.py (hunk 1 above)
Comment (KevinPlus3) [inline src/watchmaker/managers/base.py]: Okay. Sounds like a plan.

**Comment by lorengordon** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/plus3it/watchmaker/43)*

**Pull request merged**